MessageApp/Core Layer/NewCoreDataStack.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  NewCoreDataStack.swift
+//  MessageApp
+//
+//  Created by Macbook on 17.10.2022.
+//
+
+import CoreData
+
+protocol ICoreDataStack {
+    func fetch<T>(fetchRequest: NSFetchRequest<T>) -> [T]?
+    func performSave(_ block: @escaping (NSManagedObjectContext) -> Void)
+}
+
+final class NewCoreDataStack: ICoreDataStack {
+    
+    private lazy var container: NSPersistentContainer = {
+       let container = NSPersistentContainer(name: "CoreDataModel")
+        container.loadPersistentStores { _, error in
+            if let error = error {
+                Logger.shared.message(error.localizedDescription)
+            }
+        }
+        return container
+    }()
+    
+    func fetch<T>(fetchRequest: NSFetchRequest<T>) -> [T]? {
+        return try? container.viewContext.fetch(fetchRequest)
```

Comment:
Можно убирать `return` когда сразу возвращаешь
```suggestion
        try? container.viewContext.fetch(fetchRequest)
```